CMakeLists.txt

```
@@ -0,0 +1,112 @@
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+
+project(MerryMem
```

`MerryMem` or `merry_mem`? We should be consistent. I weakly vote for the vote the latter (`merry_mem`) because as a Pythonista, how could I not vote for `snake_case` 😆 But in all seriousness, it's because not all platforms support capital letters, so when I talk about this project on LinkedIn, I want the name to look good: `merry_mem` vs `merrymem`.

Anyways, idrc as long as we're consistent.

```suggestion
project(merry_mem
```

---

Oddly enough, PascalCase seems to be the convention for CMake top-level projects. Most packages are called things like "Threads", "Vulkan", "Boost", "OpenGL". I have seen this deviate for sub-projects though (e.g., Boost calls their algorithms project boost_algorithm).

For that reason, I'm lightly contending your suggestion but I don't really mind.

---

Yeah, I figured it's standard. It's just I deal with a fairly large ROS2 code base where the project name may change in various places. Tracking down the right name is just a pain, so I would much prefer everything to be the same.

That way, it makes ctrl+F trivial too.